compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -3489,6 +3489,11 @@ private BLangExpression createConditionForMatchPattern(BLangMatchPattern matchPa
     }
 
     private BLangExpression createConditionForWildCardMatchPattern(BLangWildCardMatchPattern wildCardMatchPattern) {
+        BLangExpression expr = wildCardMatchPattern.matchExpr;
+        if (!wildCardMatchPattern.isLastPattern && expr != null) {
+            return createIsAssignableExpression(wildCardMatchPattern.pos,
```

Comment:
Shall we use `createTypeCheckExpr` instead of this. I think this is an old construct we used to have before the type-test expression. In the BirGen phase this construct is also  converted into a type-test expression.